### PR TITLE
Avoid allocating new strings every time we sign a new web request

### PR DESCRIPTION
**Pull request by AnsisMalins**
## What does this PR change?

Use a static array of prebuilt strings instead of creating new x-identity-auth-chain- strings every time we create a web request.

## How to test the changes?

Play happy paths.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: Explorer/Assets/DCL/WebRequests/RequestEnvelope.cs [hunk 1]
```diff
@@ -22,8 +22,17 @@ namespace DCL.WebRequests
         private readonly WebRequestSignInfo? signInfo;
         private readonly ISet<long>? responseCodeIgnores;
 
+        private static readonly string[] AUTH_CHAIN_HEADER_NAMES;
         private const string NONE = "NONE";
 
+        static RequestEnvelope()
+        {
+            int maxAuthChainHeaders = Enum.GetNames(typeof(AuthLinkType)).Length;
```
Comment (AnsisMalins) [inline Explorer/Assets/DCL/WebRequests/RequestEnvelope.cs:30]: Could instead add a terminating COUNT enum value to the enum and consult that.

**Comment by AnsisMalins** [comment]
Blarg, the "Verified" is now showing up.

**Review by m3taphysics — APPROVED** [review]
Awesome!

**Comment by AnsisMalins** [comment]
Finally got the commit verified. Also, I realized a mistake in my original solution so I changed it. You'll have to take a look again.

**Review by m3taphysics — APPROVED** [review]

**Comment by DafGreco** [comment]
Checked post merge on windows and macOS ✅ 

Sanity check performed :
- Emotes 
- Backpack
- On boarding 
- Teleporting
- New user login 
- Metadynelabs

**Pull request merged**